agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -801,19 +801,13 @@ bool monitor_thread::handle_cmdu_vs_message(Socket &sd, ieee1905_1::CmduMessageR
         }
 
         auto sta_node = mon_db.sta_find(sta_mac);
-        if (sta_node) {
-            mon_db.sta_erase(sta_mac);
-        }
-
-        auto vap_node = mon_db.vap_get_by_id(vap_id);
-        if (vap_node == nullptr) {
-            LOG(ERROR) << "vap_id " << vap_id << " doesn't not exists";
+        if (!sta_node) {
```

Comment:
So, due to the race condition I mentioned, the sta_node may actually not exists yet.

Or, it is possible that the STA has already disconnected by the time this message arrives. In the latter case, responding with error is actually OK... Meh, race conditions are hard...